File: app/src/main/java/com/example/myapplication/ViewModel/CartViewModel.java
package com.example.myapplication.ViewModel;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.myapplication.NetworkService;

import java.util.ArrayList;

import Models.linked.Product;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
public class CartViewModel extends ViewModel {
    MutableLiveData<ArrayList<Product>> CartLiveData;
    public CartViewModel()
    {
        CartLiveData= new MutableLiveData<>();
        init();
    }
    public MutableLiveData<ArrayList<Product>> GetCartLiveData()
    {
        return CartLiveData;
    }
    public void init()
    {
        NetworkService.getInstance().getApi().GetShoppingCart().enqueue(new Callback<ArrayList<Product>>() {
            @Override
            public void onResponse(Call<ArrayList<Product>> call, Response<ArrayList<Product>> response) {
                CartLiveData.postValue(response.body());
            }
            @Override
            public void onFailure(Call<ArrayList<Product>> call, Throwable t) {
            }
        });
    }
}

File: app/src/main/java/com/example/myapplication/ViewModel/CartViewModel.kt
package com.example.myapplication.ViewModel

import com.example.myapplication.NetworkService.Companion.instance
import androidx.lifecycle.ViewModel
import androidx.lifecycle.MutableLiveData
import Models.linked.Product
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.util.ArrayList

class CartViewModel : ViewModel() {
    var CartLiveData: MutableLiveData<ArrayList<Product>>
    fun GetCartLiveData(): MutableLiveData<ArrayList<Product>> {
        return CartLiveData
    }

    fun init() {
        instance!!.api.GetShoppingCart()?.enqueue(object : Callback<ArrayList<Product>?> {
            override fun onResponse(
                call: Call<ArrayList<Product>?>,
                response: Response<ArrayList<Product>?>
            )
            {
                CartLiveData.postValue(response.body())
            }

            override fun onFailure(call: Call<ArrayList<Product>?>, t: Throwable) {}
        })
    }
    init {
        CartLiveData = MutableLiveData()
        init()
    }
}
